utils/session.c

```diff
@@ -498,11 +498,13 @@ static void delete_task(struct uftrace_task *t)
 {
 	struct uftrace_sess_ref *sref, *tmp;
 
-	sref = t->sref.next;
-	while (sref) {
-		tmp = sref->next;
-		free(sref);
-		sref = tmp;
+	if (t->sref_last) {
+		sref = t->sref.next;
+		while (sref) {
+			tmp = sref->next;
+			free(sref);
+			sref = tmp;
+		}
```

Comment:
I'd rather zero-ing the `t->sref` in `create_task()` when `needs_session` is false.

Reply:
@namhyung I fixed it according to your advice.

Reply:
Hello @namhyung , There seems to be a problem here.
Hmm... There is no record of file changes(delete_task()) in the commit history.

https://github.com/namhyung/uftrace/commits/aaa4c0c576c81989c110405a7b1a371d80942a7a/utils/session.c

Reply:
@gy741 it's not needed since we init `t->sref.next` when session is not used (in case `t->sef_last == NULL`).

Reply:
@namhyung  Hmm...
However, the segmentation fault continues to occur.
Is this a new issue?

gdb log:
```
karas$:~/uftrace/build/bin$ gdb ./uftrace
(gdb) r info
Starting program: /home/karas/uftrace/build/bin/uftrace info
warning: the debug information found in "/lib64/ld-2.26.so" does not match "/lib64/ld-linux-x86-64.so.2" (CRC mismatch).

[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".

Program received signal SIGSEGV, Segmentation fault.
0x00005555555818ba in delete_task (t=0x5555557ec0a0) at /home/karas/uftrace/utils/session.c:507
507			tmp = sref->next;
(gdb) bt full
#0  0x00005555555818ba in delete_task (t=0x5555557ec0a0) at /home/karas/uftrace/utils/session.c:507
        sref = 0x2f7a6c6d2f736172
        tmp = 0x5555557ec0c0
#1  0x0000555555581944 in delete_tasks (sessions=0x7fffffffdd58) at /home/karas/uftrace/utils/session.c:524
        t = 0x5555557ec0a0
        n = 0x5555557ec0c0
#2  0x000055555558138c in delete_sessions (sessions=0x7fffffffdd58) at /home/karas/uftrace/utils/session.c:362
        sess = 0x7fffffffdc00
        n = 0x7ffff6d800e8
#3  0x0000555555587c23 in close_data_file (opts=0x7fffffffde40, handle=0x7fffffffdc00) at /home/karas/uftrace/utils/data-file.c:539
No locals.
#4  0x000055555557a710 in command_info (argc=0, argv=0x7fffffffe078, opts=0x7fffffffde40) at /home/karas/uftrace/cmds/info.c:1137
        ret = 0
        handle = {fp = 0x5555557ebe70, sock = 0, dirname = 0x5555555b9486 "uftrace.data", hdr = {magic = "Ftrace!", version = 4, 
            header_size = 40, endian = 1 '\001', class = 2 '\002', feat_mask = 611, info_mask = 15359, max_stack = 1024, unused1 = 0, 
            unused2 = 0}, info = {exename = 0x5555557ed0b0 "/home/karas/uftrace/build/bin/a.out", 
            build_id = "b\330\364*l\343+Qk\261\033^\274\272\365\356Z\330'M", exit_status = 0, 
            cmdline = 0x5555557ed0f0 "./uftrace record a.out", nr_cpus_online = 1, nr_cpus_possible = 1, 
            cpudesc = 0x5555557ed110 "Intel(R) Core(TM) i7-7500U CPU @ 2.70GHz", meminfo = 0x5555557ed150 "1.4 / 7.6 GB (free / total)", 
            kernel = 0x5555557ed180 "Linux 4.13.0-46-generic", hostname = 0x5555557ed1b0 "karas-sec", 
            distro = 0x5555557ed1d0 "\"Ubuntu 17.10\"", argspec = 0x0, retspec = 0x0, autoarg = 0x0, autoret = 0x0, autoenum = 0x0, 
            auto_args_enabled = false, nr_tid = 1, tids = 0x5555557ed270, stime = 0, utime = 0.0050039999999999998, 
            record_date = 0x5555557ed290 "Thu Sep  6 16:33:01 2018", elapsed_time = 0x5555557ed2c0 "0.005477841 sec", vctxsw = 1, 
            ictxsw = 10, maxrss = 3792, major_fault = 0, minor_fault = 258, rblock = 0, wblock = 8, load1 = 1.33000004, 
            load5 = 1.07000005, load15 = 0.839999974, patt_type = PATT_REGEX, 
            uftrace_version = 0x5555557ed2e0 "v0.8.3-404-gaaa4 ( dwarf python tui perf sched )"}, kernel = 0x0, perf = 0x0, tasks = 0x0, 
          sessions = {root = {rb_node = 0x5555557ec130}, tasks = {rb_node = 0x0}, first = 0x5555557ec130}, nr_tasks = 0, nr_perf = 0, 
          last_perf_idx = -1, depth = 1024, needs_byte_swap = false, needs_bit_swap = false, time_filter = 0, time_range = {first = 0, 
            start = 0, stop = 0, start_elapsed = false, stop_elapsed = false, kernel_skip_out = true, event_skip_out = true}, events = {
            prev = 0x7fffffffddb0, next = 0x7fffffffddb0}}
#5  0x000055555555f6b5 in main (argc=0, argv=0x7fffffffe078) at /home/karas/uftrace/uftrace.c:1056
        opts = {lib_path = 0x0, filter = 0x0, trigger = 0x0, tid = 0x0, exename = 0x0, dirname = 0x5555555b9486 "uftrace.data", 
          logfile = 0x0, host = 0x0, sort_keys = 0x0, args = 0x0, retval = 0x0, diff = 0x0, fields = 0x0, patch = 0x0, event = 0x0, 
          run_cmd = 0x0, opt_file = 0x0, script_file = 0x0, diff_policy = 0x0, mode = 5, idx = 2, depth = 1024, kernel_depth = 0, 
          max_stack = 1024, port = 8090, color = 1, column_offset = 8, sort_column = 2, nr_thread = 0, rt_prio = 0, bufsize = 131072, 
          kernel_bufsize = 0, threshold = 0, sample_time = 0, flat = false, libcall = true, print_symtab = false, force = false, 
          report_thread = false, no_merge = false, nop = false, time = false, backtrace = false, use_pager = true, avg_total = false, 
          avg_self = false, disabled = false, report = false, column_view = false, want_bind_not = false, task_newline = false, 
          chrome_trace = false, comment = true, flame_graph = false, libmcount_single = false, kernel = false, kernel_skip_out = true, 
          kernel_only = false, keep_pid = false, list_event = false, event_skip_out = true, no_event = false, nest_libcall = false, 
          record = false, auto_args = false, libname = false, no_randomize_addr = false, range = {first = 0, start = 0, stop = 0, 
            start_elapsed = false, stop_elapsed = false, kernel_skip_out = true, event_skip_out = true}, patt_type = PATT_REGEX}
        argp = {options = 0x5555557d6040 <uftrace_options>, parser = 0x55555555d979 <parse_option>, 
          args_doc = 0x5555555b9320 "[record|replay|live|report|info|dump|recv|graph|script|tui] [<program>]", 
          doc = 0x5555555b9368 "uftrace -- function (graph) tracer for userspace", children = 0x0, help_filter = 0x0, argp_domain = 0x0}
        ret = -1
        __func__ = "main"
```

Reply:
The original patch had it (now removed).

Reply:
Did you test with the current master?  What's the version?

Note that #469 is not solved yet and it still needs your patch..

Reply:
I am using current master and using a patch made by honggyukim. #469.